client/setec/store.go

```diff
@@ -349,22 +377,55 @@ func StaticFile(path string) (Secret, error) {
 	return func() []byte { return bs }, nil
 }
 
+// hasExpired reports whether cs is an undeclared secret whose last access time
+// was longer ago than the expiry window.
+func (s *Store) hasExpired(cs *cachedSecret) bool {
+	if cs.Declared {
+		return false // declared secrets do not expire
+	} else if s.expiryAge <= 0 {
+		return false // no expiry age is defined
+	}
+	age := time.Unix(s.timeNow(), 0).UTC().Sub(cs.lastAccessTime())
+	return age > s.expiryAge
+}
+
+// snapshotActive captures a point-in-time snapshot of the active names and
+// versions of all secrets known to the store. This permits an update poll to
+// do the time-consuming lookups outside the lock.
+func (s *Store) snapshotActive() map[string]api.SecretVersion {
+	s.active.RLock()
+	defer s.active.RUnlock()
+	m := make(map[string]api.SecretVersion)
+	for name, cs := range s.active.m {
+		if s.hasExpired(cs) {
+			m[name] = 0 // 0 means "expired"
```

Comment:
`updates` is the way the poller communicates what needs to be updated, but now "updates" could be either new secret values (as before) or expirations (new).

Since this is all internal to the type, I didn't think it was worth defining a whole wrapper type to stick in a type code and a value, but we could do that.

Reply:
I added a type with named fields to make it easier to read; I left the `updates` mechanism alone, but added more documentation to it.